### PR TITLE
Companion to #14183: FRAME: Allow message ID to be mutated in `ProcessMessage`

### PR DESCRIPTION
**Pull request by gavofyork**
Just makes compatible with a minor change to the function signature of `ProcessMessage` trait.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2865886

**Review by ggwpez — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -210,6 +210,265 @@ runtime-metrics = [ "polkadot-cli/runtime-metrics" ]
 pyroscope = ["polkadot-cli/pyroscope"]
 jemalloc-allocator = ["polkadot-node-core-pvf-worker/jemalloc-allocator", "polkadot-overseer/jemalloc-allocator"]
 
+[patch."https://github.com/paritytech/substrate"]
+node-template = { path = "/Users/gav/Core/substrate/bin/node-template/node" }
```
Comment (ggwpez) [inline Cargo.toml:214]: You can revert this, the merge bot will then update the lock (it is expected not to build here).

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2865943

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/2865987

**Review by sam0x17 — APPROVED** [review]

**Review by bkchr — APPROVED** [review]

**Pull request merged**